astropy/cosmology/tests/test_core.py

```diff
@@ -0,0 +1,325 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Testing :mod:`astropy.cosmology.core`."""
+
+##############################################################################
+# IMPORTS
+
+import abc
+import inspect
+from types import MappingProxyType
+
+import pytest
+
+import numpy as np
+
+import astropy.units as u
+from astropy.cosmology import Cosmology, core
+from astropy.cosmology.core import _COSMOLOGY_CLASSES, Parameter
+
+##############################################################################
+# TESTS
+##############################################################################
+
+
+class TestParameter:
+    """Test `astropy.cosmology.Parameter`."""
+
+    def setup_class(self):
+        class Example1(Cosmology):
+            param = Parameter(doc="example parameter",
+                              unit=u.m, equivalencies=u.mass_energy())
+
+            def __init__(self, param=15):
+                self._param = param
+
+        class Example2(Example1):
+            def __init__(self, param=15 * u.m):
```

Comment:
Why use an example that changes the unit but not the value? Maybe at least do `param.to_value(u.km)`?

Reply:
Done. Had to adjust expected output in a subsequent test.